src/Symfony/Bundle/FrameworkBundle/Debug/TraceableEventDispatcher.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Debug;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpKernel\Debug\DebugRequestStack;
+use Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher as BaseTraceableEventDispatcher;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * @internal
+ */
+final class TraceableEventDispatcher extends BaseTraceableEventDispatcher
+{
+    public function __construct(EventDispatcherInterface $dispatcher, Stopwatch $stopwatch, LoggerInterface $logger = null, private ?DebugRequestStack $requestStack = null)
+    {
+        parent::__construct($dispatcher, $stopwatch, $logger, $requestStack);
+    }
+
+    protected function beforeDispatch(string $eventName, object $event): void
+    {
+        if (!class_exists(ConsoleEvents::class)) {
+            parent::beforeDispatch($eventName, $event);
+
+            return;
+        }
+
+        switch ($eventName) {
+            case ConsoleEvents::COMMAND:
+                $request = $this->requestStack->getCurrentRequest(ignoreVirtual: false);
+
+                /** @var ConsoleCommandEvent $event */
+                if (!$request instanceof CliRequest || $request->command !== $event->getCommand()) {
+                    break;
+                }
+
+                $request->attributes->set('_stopwatch_token', substr(hash('sha256', uniqid(mt_rand(), true)), 0, 6));
+                $this->stopwatch->openSection();
```

Comment:
Not calling the parent logic means that it might not properly trace other listeners on that event if we ever add some logic in the parent method instead of being an empty method.

Reply:
Parent logic of http kernel should not be concerned by a console-managed request stack.
I don't think I understand the reasoning.
Parent call is done early and in default case.

Reply:
The parent logic is not about the request stack but about the event dispatching.

Reply:
The Event Dispatcher should be event agnostic, hence the empty method.
In the current implementation of the http kernel, no case in the switch is actually calling the parent method.
Does this imply that this old implementation should be fixed as well?